src/main/java/com/hotels/beeju/extensions/BeejuJUnitExtension.java

```
@@ -0,0 +1,145 @@
+ */
+package com.hotels.beeju.extensions;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.thrift.TException;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.hotels.beeju.core.BeejuCore;
+
+abstract public class BeejuJUnitExtension implements BeforeEachCallback, AfterEachCallback {
+
+  private static final Logger LOG = LoggerFactory.getLogger(BeejuJUnitExtension.class);
+  public BeejuCore core;
+  private Path tempDirectory;
+
+  public BeejuJUnitExtension(String databaseName, Map<String, String> configuration) {
+    core = new BeejuCore(databaseName, configuration);
+  }
+
+  /**
+   * Initialise the warehouse path.
+   * <p>
+   * This method can be overridden to provide additional initialisations.
+   * </p>
+   *
+   * @throws Throwable If the initialisation fails.
+   */
+  protected void init() throws Throwable {
+    tempDirectory = Files.createTempDirectory("root");
+    core.setHiveVar(HiveConf.ConfVars.METASTOREWAREHOUSE, tempDirectory.toString());
+  }
+
+  /**
+   * Implement this method to prepare the rule for a new test.
+   * <p>
+   * This is called after the rule is initialised.
+   * </p>
+   *
+   * @throws Throwable If the rule cannot be prepared for a new test.
+   */
+  protected abstract void beforeTest() throws Throwable;
+
+  @Override
+  public void beforeEach(ExtensionContext context) throws TException {
+    try {
+      init();
+      beforeTest();
+    } catch (Throwable throwable) {
+      LOG.error("Unable to initialise BeejuJUnit extension", throwable);
```

Not really useful, imagine running a test that is not working and you only have this this error in the logs somewhere., that's hard to debug. Just remove the try catch and let all the exceptions bubble up. Fail fast